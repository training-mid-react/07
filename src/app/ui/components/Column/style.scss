.column {
  display: flex;
  flex-direction: column-reverse;
  width: 100px;
  cursor: pointer;
  border-right-style: solid;
  border-left-style: solid;
  border-bottom-style: solid;
  border-color: black;
}