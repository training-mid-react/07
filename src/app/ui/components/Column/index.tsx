import './style.scss';
import { useDropCell } from '../../../core/hooks/useDropCell';

interface ColumnProps {
  columnIndex: number;
  dropPiece: (column: number) => void;
  columnData: number[];
}

function Column({ columnIndex, dropPiece, columnData }: ColumnProps) {
  const { handleColumnClick, droppingCell } = useDropCell({ columnIndex, dropPiece, columnData })
  return (
    <div className="column" onClick={handleColumnClick}>
      {columnData.map((cell, rowIndex) => {
        const isDropping = droppingCell === rowIndex;
        const cellClass = `cell ${cell === 1 ? 'cell--player1' : cell === 2 ? 'cell--player2' : ''} ${isDropping ? 'cell--dropping' : ''}`;
        return (
          <div
            key={rowIndex}
            className={cellClass}
          />
        );
      })}
    </div>
  );
};

export default Column;
