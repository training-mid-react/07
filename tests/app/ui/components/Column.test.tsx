import { render } from "@testing-library/react";
import { describe, expect, test, vi } from "vitest";
import React from "react";
import Column from "../../../../src/app/ui/components/Column";

describe('Pruebas en el componente Column', () => {
    test('Match snapshot', () => {
        render(<Column columnIndex={1} dropPiece={() => { }} columnData={[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]} />)
        expect(document.body).toMatchSnapshot();
    })
    test('Match snapshot', () => {
        render(<Column columnIndex={2} dropPiece={() => { }} columnData={[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]} />)
        expect(document.body).toMatchSnapshot();
    })
    test('debe tener un data-testid Ãºnico para la columna', () => {
        const mockBoard = [
            [0, 0, 0, 0, 0, 0, 0],
            [0, 0, 0, 0, 0, 0, 0],
            [0, 0, 0, 0, 0, 0, 0],
            [0, 0, 0, 0, 0, 0, 0],
            [0, 0, 0, 0, 0, 0, 0],
            [0, 0, 0, 0, 0, 0, 0],
        ];

        const handleDropPiece = vi.fn();
        const { getByTestId } = render(
            <Column columnIndex={0} dropPiece={handleDropPiece} columnData={mockBoard[0]} />
        );

        const columnElement = getByTestId('column-0');
        expect(columnElement).toBeInTheDocument();
    });

    test('debe tener las clases correctas para cada celda dependiendo de su valor', () => {
        const mockBoard = [
            [1, 2, 0, 0, 0, 0, 0],
        ];

        const handleDropPiece = vi.fn();
        const { getByTestId } = render(
            <Column columnIndex={0} dropPiece={handleDropPiece} columnData={mockBoard[0]} />
        );

        const cell0 = getByTestId('cell-0-0');
        const cell1 = getByTestId('cell-0-1');
        const cell2 = getByTestId('cell-0-2');

        expect(cell0).toHaveClass('cell--player1');
        expect(cell1).toHaveClass('cell--player2');
        expect(cell2).not.toHaveClass('cell--player1');
        expect(cell2).not.toHaveClass('cell--player2');
    });

});