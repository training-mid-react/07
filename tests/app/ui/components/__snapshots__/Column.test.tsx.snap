// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pruebas en el componente Column > Match snapshot 1`] = `
<body>
  <div>
    <div
      class="column"
      data-testid="column-1"
    >
      <div
        class="cell cell--player1 "
        data-testid="cell-1-0"
      />
      <div
        class="cell cell--player2 "
        data-testid="cell-1-1"
      />
      <div
        class="cell  "
        data-testid="cell-1-2"
      />
      <div
        class="cell  "
        data-testid="cell-1-3"
      />
      <div
        class="cell  "
        data-testid="cell-1-4"
      />
      <div
        class="cell  "
        data-testid="cell-1-5"
      />
      <div
        class="cell  "
        data-testid="cell-1-6"
      />
      <div
        class="cell  "
        data-testid="cell-1-7"
      />
      <div
        class="cell  "
        data-testid="cell-1-8"
      />
      <div
        class="cell  "
        data-testid="cell-1-9"
      />
    </div>
  </div>
</body>
`;

exports[`Pruebas en el componente Column > Match snapshot 2`] = `
<body>
  <div>
    <div
      class="column"
      data-testid="column-2"
    >
      <div
        class="cell cell--player1 "
        data-testid="cell-2-0"
      />
      <div
        class="cell cell--player2 "
        data-testid="cell-2-1"
      />
      <div
        class="cell  "
        data-testid="cell-2-2"
      />
      <div
        class="cell  "
        data-testid="cell-2-3"
      />
      <div
        class="cell  "
        data-testid="cell-2-4"
      />
      <div
        class="cell  "
        data-testid="cell-2-5"
      />
      <div
        class="cell  "
        data-testid="cell-2-6"
      />
      <div
        class="cell  "
        data-testid="cell-2-7"
      />
      <div
        class="cell  "
        data-testid="cell-2-8"
      />
      <div
        class="cell  "
        data-testid="cell-2-9"
      />
    </div>
  </div>
</body>
`;
